functionary/core/utils/messaging.py

```diff
@@ -0,0 +1,139 @@
+import json
+import logging
+import ssl
+from typing import Tuple
+
+import pika
+from django.conf import settings
+from pika.exceptions import UnroutableError
+from pika.exchange_type import ExchangeType
+
+logger = logging.getLogger(__name__)
+
+PUBLIC_EXCHANGE = "runners.public"
+PUBLIC_QUEUE = "public"
+TASK_RESULTS_QUEUE = "tasking.results"
+
+
+def build_connection(ca=None, cert=None, key=None, open_callback=None):
```

Comment:
There's a nearly identical version of this in the runner code.  I'm going to live with that for now and maybe at some point figure out how to consolidate these things.